synapse/handlers/sso.py

```diff
@@ -1232,6 +1232,91 @@ async def revoke_sessions_for_provider_session_id(
             )
             await self._device_handler.delete_devices(user_id, [device_id])
 
+    async def invalidate_sessions_for_provider_session_id(
```

Comment:
This feels like it should be instead a parameter on the `revoke_sessions_for_provider_session_id` method instead, because most of the logic here is duplicated

Reply:
I suspect it would *improve* readability if there's less duplicated code, so we can see what the differences are based on the flag.

Reply:
@hachem2001 Do you have interest in making this change?